client/login/magic-login/handle-emailed-link-form.jsx

```diff
@@ -40,42 +76,75 @@ class HandleEmailedLinkForm extends React.Component {
 		this.props.fetchMagicLoginAuthenticate( this.props.emailAddress, this.props.token, this.props.tokenTime );
 	};
 
-	componentWillMount() {
-		const { emailAddress, token, tokenTime } = this.props;
-
-		if ( emailAddress && emailValidator.validate( emailAddress ) && token && tokenTime ) {
-			return;
+	// Lifted from `blocks/login`
+	// @TODO unify
+	handleValidToken = () => {
+		if ( ! this.props.twoFactorEnabled ) {
+			this.rebootAfterLogin();
+		} else {
+			page( login( {
+				isNative: true,
+				// If no notification is sent, the user is using the authenticator for 2FA by default
+				twoFactorAuthType: this.props.twoFactorNotificationSent.replace( 'none', 'authenticator' )
+			} ) );
 		}
+	};
 
-		this.props.showMagicLoginLinkExpiredPage();
-	}
+	// Lifted from `blocks/login`
+	// @TODO unify
+	rebootAfterLogin = () => {
+		const { redirectTo } = this.props;
+
+		this.props.recordTracksEvent( 'calypso_login_success', {
+			two_factor_enabled: this.props.twoFactorEnabled,
+			magic_login: 1
```

Comment:
As https://github.com/Automattic/eslint-config-wpcalypso/pull/5 should be merged soon, I recommend adding a trailing comma there.